libs/ledgerjs/packages/hw-app-eth/src/services/ledger/erc20.ts

```diff
@@ -35,11 +60,12 @@ const asContractAddress = (addr: string) => {
 };
 
 // this internal get() will lazy load and cache the data from the erc20 data blob
-const get: () => API = (() => {
-  let cache;
-  return () => {
-    if (cache) return cache;
-    const buf = Buffer.from(blob, "base64");
+const get: (erc20SignaturesBlob?: string) => API = (() => {
+  const wm: WeakMap<{ blob: string | undefined }, API> = new WeakMap();
+  return (erc20SignaturesBlob) => {
+    if (wm.has({ blob: erc20SignaturesBlob }))
```

Comment:
i don't think it works, because you recreate an object reference, `weakmap.has({ })` is always returning false because `{}` is a new object when you inline it.

you can however just have `erc20SignaturesBlob ` as a key?

Reply:
I tried but 
```
Interface WeakMap<K, V>
Type parameters
K: object
V
```

K is an object here unfortunately, I'll try something else

Reply:
I tried to do a cache map instead, my way to use weakmap doesn't seems to work well. In fact that didn't cache anything